gui-themes/themes/zeit-amp/scripts/js/plugins/ampify.js

```diff
@@ -5,35 +5,28 @@ define([
 ], function(plugins, utils, dust) {
     'use strict';
 
-    plugins['ampify'] = function(config) {
+    plugins.ampify = function(config) {
 
         utils.dispatcher.on('after-render.post-view', function(view) {
-            view.$( '*[style]' ).removeAttr( 'style' );
-
-            switch ( view.el.className ) {
-                case 'image':
-                    view.$( '.post-external-image img' ).each( function() {
-                        var self = this,
-                            img = new Image();
-                        img.src = this.src;
-                        img.onload = function() {
-                            view.$( self ).replaceWith( '<amp-img src="' + this.src + '" width="' + this.width + '" height="' + this.height + '" layout="responsive"></amp-img>' );
-                            view.$( img ).remove();
-                        };
-                        view.$el.append( img );
-                    });
-                    break;
+            view.$('*[style]').removeAttr('style');
 
+            switch (view.$el.attr('class')) {
                 case 'link':
-                    view.$( '.link-thumbnail img' ).replaceWith( function() {
-                        return '<amp-img src="' + this.src + '" width="' + this.width + '" height="' + this.height + '" layout="responsive"></amp-img>';
-                    });
-                    view.$( '.source-icon img' ).replaceWith( function() {
-                        return '<amp-img src="' + this.src + '" width="16" height="16" layout="fixed"></amp-img>';
-                    });
+                    var icon = view.$('.source-icon img');
+                    icon.replaceWith('<amp-img src="' + icon.attr('src') + '" width="16" height="16" layout="fixed"></amp-img>');
                     break;
             }
 
+            var images = view.$('img'),
+                img, width, height, index;
+
+            for (index = images.length; index--;) {
+                // @TODO get width and height of image if missing
+                img = images.eq(index);
+                width = img.attr('width') || 400;
+                height = img.attr('height') || 300;
+                img.replaceWith('<amp-img src="' + img.attr('src') + '" width="' + width + '" height="' + height + '" layout="responsive"></amp-img>');
+            }
```

Comment:
I changed quite a lot to get it working in embed server, but image `width` and `height` is still an issue.